File: frontend/lib/models/auth_response.dart
// auth_response.dart — Parses login and token refresh API responses
import 'package:flutter/foundation.dart';

/// Lightweight user info embedded in the login/refresh response.
///
/// Full profile details are fetched separately via GET /api/v1/profile.
@immutable
class AuthUser {
  final int    id;
  final String fullName;
  final String role;

  const AuthUser({
    required this.id,
    required this.fullName,
    required this.role,
  });

  factory AuthUser.fromJson(Map<String, dynamic> json) => AuthUser(
        id:       json['id'],
        fullName: json['fullName'],
        role:     json['role'],
      );
}

/// Full response from POST /auth/login and POST /auth/refresh.
@immutable
class AuthResponse {
  final String   accessToken;
  final String   refreshToken;
  final String   tokenType;
  final int      expiresIn;
  final AuthUser user;

  const AuthResponse({
    required this.accessToken,
    required this.refreshToken,
    required this.tokenType,
    required this.expiresIn,
    required this.user,
  });

  factory AuthResponse.fromJson(Map<String, dynamic> json) => AuthResponse(
        accessToken:  json['accessToken'],
        refreshToken: json['refreshToken'],
        tokenType:    json['tokenType']  ?? 'Bearer',
        expiresIn:    json['expiresIn']  ?? 3600,
        user:         AuthUser.fromJson(json['user'] as Map<String, dynamic>),
      );
}

File: backend/src/main/java/com/smartpark/dto/response/AuthResponse.java
// Response DTO for login and refresh endpoints — contains JWT tokens and basic user info.
package com.smartpark.dto.response;

public class AuthResponse {

    private String accessToken;
    private String refreshToken;
    private String tokenType;
    private long expiresIn;
    private UserInfo user;

    public AuthResponse(String accessToken, String refreshToken, UserInfo user) {
        this.accessToken = accessToken;
        this.refreshToken = refreshToken;
        this.tokenType = "Bearer";
        this.expiresIn = 3600;
        this.user = user;
    }

    public String getAccessToken() { return accessToken; }
    public String getRefreshToken() { return refreshToken; }
    public String getTokenType() { return tokenType; }
    public long getExpiresIn() { return expiresIn; }
    public UserInfo getUser() { return user; }

    public static class UserInfo {
        private Long id;
        private String fullName;
        private String role;

        public UserInfo(Long id, String fullName, String role) {
            this.id = id;
            this.fullName = fullName;
            this.role = role;
        }

        public Long getId() { return id; }
        public String getFullName() { return fullName; }
        public String getRole() { return role; }
    }
}